src/openrct2/scripting/bindings/ride/ScRide.cpp

```diff
@@ -524,6 +524,72 @@ namespace OpenRCT2::Scripting
         return ride != nullptr ? ride->GetRideTypeDescriptor().LiftData.minimum_speed : 0;
     }
 
+    int32_t ScRide::maxSpeed_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ((ride->max_speed) * 9) >> 18 : 0;
+    }
+
+    int32_t ScRide::averageSpeed_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ((ride->average_speed) * 9) >> 18 : 0;
+    }
+
+    int32_t ScRide::rideTime_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ride->GetTotalTime() : 0;
+    }
+
+    int32_t ScRide::rideLength_get() const
+    {
+        auto ride = GetRide();
+        return ride != nullptr ? ride->GetTotalLength() >> 16 : 0;
```

Comment:
Aren't we loosing some accuracy here either?

Reply:
Return value is in meters, so we are loosing sub-meter accuracy. But it yields the value which is used in the ride stats tab, so I think this is fine.

Reply:
It makes it harder to use the values for other purposes though, like if a plugin want to mimic sheltered surfaces with the total ride length or use the `highest_drop_height` for calculating certain ride ratings. Also note that internally `highest_drop_height` is in height units, which is a perfectly valid measurement used in other parts of the plugin API.

Reply:
We're kind of running into the same issue with Max Speed and Average Speed as well, so should we not modify those either? If there are specific use cases where we can't use these values, I can make `rideLengthUnmodified_get()` functions as well.

Reply:
On of the ideas of the plugin API design is to hide how things internally work. That also means that we should not provide raw values, especially since they might me subject of change.
If higher precision is absolutely needed, I would go for milliseconds or real values.

Reply:
Or if possible, maybe add the extra detail as a floating point? Not sure how the `formatString` will interpret that though.

Reply:
That's what I said, no?

Reply:
![image](https://github.com/OpenRCT2/OpenRCT2/assets/15895532/359767d4-ec11-4b1f-b45b-30ff229493a4)
`formatString` seems to have no issue with that.

Reply:
Sounds good, I'll return these values with higher precision then, maybe into the thousandths.